test/reload_test.go

```diff
@@ -202,4 +204,132 @@ func TestReloadSeveralTimeMetrics(t *testing.T) {
 	}
 }
 
+func TestMetricsAvailableAfterReload(t *testing.T) {
+	//TODO: add a tool that find an available port because this needs to be a port
```

Comment:
If you use port 0, it should do that for you shouldn't it?

Reply:
Yes, I can .. and it is done in some cases.
But I need to know what is the port to send an HTP request ...
there is a trick right now: the opened port is in a global var "ListenAddr". But global var is not compatible with Reload, neither with these Tests that are running in parallel in the same process.

We should definitively design a solution to avoid the global var. For prod and for Tests.

Waiting a solution, the shortcut is to use a different port for each Test.

Reply:
met.Addr has the value provided in the Stanzs => "localhost:0"

Right now, the one that now the effective port associated to the listener is
https://github.com/coredns/coredns/blob/master/plugin/metrics/metrics.go#L96
```
ListenAddr = m.ln.Addr().String() // For tests
```

But it is a global var of that package .

Reply:
But we could save it in met.EffectiveAddr and use it.

Reply:
I had a look. It is useless to try to save the effective address into the metric plugins ... but it would be useful to save it into the instance.Storage that exist in caddy's instance.

We can now save any data from the Controller (in Setup) along with a Key, and access the same data using an Instance of Caddy.

But I think will be in another PR.